variable.c

```diff
@@ -214,6 +214,22 @@ is_constant_path(VALUE name)
  *    c.new # => #<C:0x0....>
  */
 
+
+static VALUE
+build_const_path(VALUE head, ID tail);
+
+static enum rb_id_table_iterator_result
+set_temporary_name_i(ID id, VALUE v, void *payload)
+{
+    rb_const_entry_t *ce = (rb_const_entry_t *)v;
+    VALUE value = ce->value;
+    VALUE parental_path = *((VALUE *) payload);
+
+    rb_mod_set_temporary_name(value, build_const_path(parental_path, id));
```

Comment:
If the anonymous module has a constant that is not anonymous class or module, this will raise an exception.
And will crash on a non-class/module constant.